### PR TITLE
feat(E2E): added new tests & support for parallel execution of E2E Tests

### PR DESCRIPTION
**Pull request by cxlucas**
Closes #

**Proposed Changes**

- Add support for parallel execution of E2E Tests, considerably reducing the duration of E2E Tests.
- Add E2E test cases that cover new flags and the text content of generated log files.

| Test Name   | Description                                                                                                                                                   | Expected Output | Exit Status Code |
|-------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------|-----------------|------------------|
| E2E-CLI-035 |                           Kics scan command with --exclude-results should not run/found results (similarityID) provided by this flag                          |   status code   |      20, 126     |
| E2E-CLI-036 |            KICS scan command with --include-queries should perform a scan running only the queries that match the query-ids provided from this flag           |   match regex   |  50, 40, 20, 126 |
| E2E-CLI-037 | KICS scan command with --exclude-results and --include-queries should run only provided queries and does not run results (similarityID) provided by this flag |   status code   |       0, 20      |
| E2E-CLI-038 |                                         KICS scan command with --log-path should perform a scan and create a log file.                                        |     stdout      |        40        |
| E2E-CLI-039 |                 KICS scan command with --log-path and --log-level should perform a scan and create a log file based in the provided log-level.                |     stdout      |        40        |

I submit this contribution under the Apache-2.0 license.

### CODE REVIEW

File: e2e/README.md [hunk 1]
```diff
@@ -164,6 +177,44 @@ Files created by Kics during testing should always be created in the 'outputs' f
 }
 ```
 
+<b>Example 4:</b> A test case that executes a command and checks: 
+- Status code output
+-   Generated <b>log file</b> content
```
Comment (felipe-avelar) [inline e2e/README.md:182]:
```suggestion
- Generated <b>log file</b> content
```

File: e2e/README.md [hunk 2]
```diff
@@ -164,6 +177,44 @@ Files created by Kics during testing should always be created in the 'outputs' f
...
+
+In contrast to what is expected for results and payload, the log files don't have a file to make the comparison, instead of that, the kics has a validation function that has the same aim as the validation that verifies the output CLI.
```
Comment (felipe-avelar) [inline e2e/README.md:184]:
```suggestion
In contrast to what is expected for results and payload, log files don't have a file to compare, instead KICS validates logs like it validates CLI's output.
```

File: e2e/README.md [hunk 3]
```diff
@@ -164,6 +177,44 @@ Files created by Kics during testing should always be created in the 'outputs' f
...
+In this case, the validation function will receive the log file data as its input, and you can do any kind of regex or text validations in the log file content.
```
Comment (felipe-avelar) [inline e2e/README.md:185]:
```suggestion
In this case, the validation function will receive the log file data as its input and you can use any regex or do text validations in the log file content.
```

File: e2e/README.md [hunk 4]
```diff
@@ -164,6 +177,44 @@ Files created by Kics during testing should always be created in the 'outputs' f
...
+
+The example below generates an output file called: <i>"E2E_CLI_038_LOG"</i> and create a validation function expecting 3 matches in the generated log file content.
```
Comment (felipe-avelar) [inline e2e/README.md:187]:
```suggestion
The example below generates an output file called: <i>"E2E_CLI_038_LOG"</i> and creates a validation function expecting 3 matches in the generated log file content.
```

File: e2e/README.md [hunk 5]
```diff
@@ -164,6 +177,44 @@ Files created by Kics during testing should always be created in the 'outputs' f
...
+
+In addition, it is necessary to remove the files that will be generated  during the test by adding them to removeFiles (required only for test cases that generates files from results/payloads/logs).
```
Comment (felipe-avelar) [inline e2e/README.md:189]:
```suggestion
In addition, it is necessary to remove files generated during the test, adding them to `removeFiles` (required only for test cases that generates files from results/payloads/logs).
```

File: e2e/README.md [hunk 6]
```diff
@@ -132,14 +145,14 @@ The custom validation function must perform a validation based on the CLI output
 <b>Example 3:</b> A test case that executes a command and checks: 
 - Status code output
 - Kics CLI output
-- Generated payload file content
+-   Generated <b>payload file</b> content
```
Comment (felipe-avelar) [inline e2e/README.md:148]:
```suggestion
- Generated <b>payload file</b> content
```

File: e2e/README.md [hunk 7]
```diff
@@ -69,10 +75,17 @@ A single test can contain verification based on:
  - status code + CLI output (file check)
  - status code + payload content (file check)
  - status code + result content (file check)
+-   status code + log content (file check with validation function)
```
Comment (felipe-avelar) [inline e2e/README.md:78]:
```suggestion
- status code + log content (file check with validation function)
```

File: e2e/README.md [hunk 8]
```diff
@@ -69,10 +75,17 @@ A single test can contain verification based on:
...
+
  - status code + result content (file check) + payload content (file check)
+-   status code + result content (file check) + payload content (file check) + log content (file check with validation function)
```
Comment (felipe-avelar) [inline e2e/README.md:81]:
```suggestion
- status code + result content (file check) + payload content (file check) + log content (file check with validation function)
```

**Review by felipe-avelar — DISMISSED** [review]
The changes looks good to me, could you check the suggestions on README file, please?

**Comment by cxlucas** [comment]
> The changes looks good to me, could you check the suggestions on README file, please?

I added the suggestions, Thanks.

**Review by rogeriopeixotocx — APPROVED** [review]

**Pull request merged**